### PR TITLE
108 extract interface database manager

### PR DESCRIPTION
**Pull request by Shtramak**
connect to #108

### CODE REVIEW

File: src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import java.sql.SQLException;
+
+public class InMemoryDatabaseManager implements DatabaseManager {
+    private Table[] tables;
+
+    public InMemoryDatabaseManager() {
+        tables = new Table[0];
+    }
+
+    public void createTable(String tableName, int colums_nember) {
```
Comment (ivshapovalov) [inline src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java]: Is it appropriate name for Java "colums_nember"?

File: src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            Table currentTable = tables[i];
+            tableNames[i] = currentTable.getTableName();
+        }
+        return tableNames;
+    }
+
+    @Override
+    public DataSet[] getTableData(String tableName) throws SQLException {
+        Table table = tableByName(tableName);
+        return table != null ? table.tableData() : new DataSet[0];
```
Comment (ivshapovalov) [inline src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java:43]: If table not exists or table not contain data - result 'getTableData' is the same?

File: src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                return currentTable;
+            }
+        }
+        return null;
+    }
+
+    private static class Table {
+        private String tableName;
+        private DataSet[] rows;
+        private int colums_nember;
```
Comment (ivshapovalov) [inline src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java]: Names wrong I think

File: src/test/java/juja/sqlcmd/InMemoryDatabaseManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
+package juja.sqlcmd;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.sql.SQLException;
+
+import static org.hamcrest.Matchers.arrayContainingInAnyOrder;
+import static org.junit.Assert.*;
```
Comment (ivshapovalov) [inline src/test/java/juja/sqlcmd/InMemoryDatabaseManagerTest.java]: '*' maximize imports

File: src/test/java/juja/sqlcmd/InMemoryDatabaseManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+import static org.junit.Assert.assertFalse;
+
+public class InMemoryDatabaseManagerTest {
+    private InMemoryDatabaseManager databaseManager;
```
Comment (ivshapovalov) [inline src/test/java/juja/sqlcmd/InMemoryDatabaseManagerTest.java]: Maybe Interface variable

**Review by ivshapovalov — CHANGES_REQUESTED** [review]

File: src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java (hunk 2 above)
Comment (Shtramak) [reply src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java:43]: Yes, it's according to the logic of getTableData() method. The same logic we have in JdbcDatabaseManager

File: src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java (hunk 1 above)
Comment (Shtramak) [reply src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java]: Yup, it's some written mistake. Thanks!

File: src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java (hunk 3 above)
Comment (Shtramak) [reply src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java]: Will be fixed

File: src/main/java/juja/sqlcmd/DatabaseManager.java [hunk 6]
```diff
@@ -1,121 +1,17 @@
...
-import java.sql.Connection;
-import java.sql.DriverManager;
-import java.sql.ResultSet;
-import java.sql.ResultSetMetaData;
 import java.sql.SQLException;
-import java.sql.Statement;
 
-public class DatabaseManager {
-    private static final String DB_CONNECTION_URL = "jdbc:postgresql://127.0.0.1:5432/";
+public interface DatabaseManager {
```
Comment (adaleorion) [inline src/main/java/juja/sqlcmd/DatabaseManager.java:5]: Please add method for update operation. 
Yes, we are going to change parameters for this method,  but without this case we will forget about it

File: src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    private Table tableByName(String tableName) {
+        for (Table currentTable : tables) {
+            if (currentTable.getTableName().equals(tableName)) {
+                return currentTable;
+            }
+        }
+        return null;
+    }
+
+    private static class Table {
```
Comment (adaleorion) [inline src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java:77]: This implementation has many operation arraycopy
I think, we could repeat idea from ArrayList implementation with initial capacity.

You can change it if you want but  for fun only :)

File: src/main/java/juja/sqlcmd/DatabaseManager.java (hunk 6 above)
Comment (Shtramak) [reply src/main/java/juja/sqlcmd/DatabaseManager.java:5]: OK. Will be done

File: src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java (hunk 7 above)
Comment (Shtramak) [reply src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java:77]: Yup! You're right, I was thinking about that, but decided to make this implementation easier... I even was thinking to provide user identification))) Maybe in future. That's interesting. Thanks for your comments!

**Review by ivshapovalov — APPROVED** [review]

**Review by adaleorion — APPROVED** [review]

**Pull request merged**